### PR TITLE
[Feature] Support Binary Data Type

### PR DESCRIPTION
**Pull request by LiShuMing**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 1]
```diff
@@ -48,6 +48,7 @@ BACKUP: 'BACKUP';
 BEGIN: 'BEGIN';
 BETWEEN: 'BETWEEN';
 BIGINT: 'BIGINT';
+BINARY: 'BINARY';
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:51]: should be nonReserved

Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:51]: fixed.

File: docs/sql-reference/sql-statements/data-types/BINARY.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+BINARY(M)
+
+VARBINARY(M)
+
+1. Since StarRocks version 3.0, StarRocks supports `BINARY`/`VARBINARY`, the maximum supported length is the same as `VARCHAR` type, and the value range of `M` is 1~1048576.
+2. `BINARY` is just an alias of `VARBINARY`, the usage is exactly the same as `VARBINARY`;
+3. `BINARY(M)`/`VARBINARY(M)` will not pad the unaligned length;
+
+## 示例
```
Comment (mofeiatwork) [inline docs/sql-reference/sql-statements/data-types/BINARY.md:14]: example ?

File: docs/sql-reference/sql-statements/data-types/BINARY.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
++-------+-----------+------+-------+---------+-------+
+2 rows in set (0.01 sec)
+
+```
+
+### Import data and store it as BINARY type
+
+StarRocks supports the following three ways to import data and store it as BINARY type.
+
+- Method 1: Use `INSERT INTO` to write data to a constant column of BINARY type (such as column `j`), where the constant column is prefixed with `x'``.
```
Comment (mofeiatwork) [inline docs/sql-reference/sql-statements/data-types/BINARY.md:49]: how to load binary data type from parquet ?

File: docs/sql-reference/sql-statements/data-types/BINARY.md (hunk 2 above)
Comment (LiShuMing) [reply docs/sql-reference/sql-statements/data-types/BINARY.md]: fix it.

File: docs/sql-reference/sql-statements/data-types/BINARY.md (hunk 3 above)
Comment (LiShuMing) [reply docs/sql-reference/sql-statements/data-types/BINARY.md:49]: This PR is only for StorRocks's inner table usage. Can I add the external table(parquet)'s binary type in the later pr? 

I know parquet's binary type should be supported yet.

Comment (mofeiatwork) [reply docs/sql-reference/sql-statements/data-types/BINARY.md:49]: Not the external table, but the broker load from parquet file

Comment (LiShuMing) [reply docs/sql-reference/sql-statements/data-types/BINARY.md:49]: Thanks for your replies. Update docs. PTAL

**Review by murphyatwork — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**